Code/Framework/AzToolsFramework/Tests/Prefab/Instance/InstanceDeserializationTests.cpp

```diff
@@ -0,0 +1,205 @@
+        EXPECT_TRUE(instantiatedPrefab != nullptr);
+
+        instantiatedPrefab->GetEntities(
+            [](const AZStd::unique_ptr<AZ::Entity>& entity)
+            {
+                // Activate the entities so that we can later validate that entities stay activated throughout the deserialization.
+                entity->Init();
+                entity->Activate();
+                return true;
+            });
+
+        return AZStd::make_pair<InstanceUniquePointer, InstanceUniquePointer>(AZStd::move(createdPrefab), AZStd::move(instantiatedPrefab));
+    }
+
+    TEST_F(InstanceDeserializationTest, ReloadInstanceUponComponentUpdate)
+    {
+        AZ::Entity* entity1 = CreateEntity("Entity1",false);
+        AzToolsFramework::Components::TransformComponent* transformComponent = aznew AzToolsFramework::Components::TransformComponent;
+        transformComponent->SetWorldTranslation(AZ::Vector3(10.0, 0, 0));
+        entity1->AddComponent(transformComponent);
+
+        AZ::Entity* entity2 = CreateEntity("Entity2", false);
+
+        AZStd::pair<InstanceUniquePointer, InstanceUniquePointer> prefabInstances =
+            SetupPrefabInstances(AzToolsFramework::EntityList{ entity1, entity2}, m_prefabSystemComponent);
+        InstanceUniquePointer createdPrefab = AZStd::move(prefabInstances.first);
+        InstanceUniquePointer instantiatedPrefab = AZStd::move(prefabInstances.second);
+
+        createdPrefab->GetEntities(
+            [](const AZStd::unique_ptr<AZ::Entity>& entity)
+            {
+                if (entity->GetName() == "Entity1")
+                {
+                    // Activate the entity to access the transform interface and use it to modify the transform component.
+                    entity->Init();
+                    entity->Activate();
+                    AZ::TransformBus::Event(entity->GetId(), &AZ::TransformInterface::SetWorldX, 20.0f);
+                }
+                return true;
+            });
+        
+        
+        GenerateDomAndReloadInstantiatedPrefab(createdPrefab, instantiatedPrefab);
+
+        instantiatedPrefab->GetEntities(
+            [](const AZStd::unique_ptr<AZ::Entity>& entity)
+            {
+                if (entity->GetName() == "Entity2")
+                {
+                    // Since we didn't touch entity2 in createdPrefab, it should remain untouched in instantiatedPrefab and thus retain its
+                    // active state.
+                    EXPECT_EQ(entity->GetState(), AZ::Entity::State::Active);
+                }
+                else
+                {
+                    // Validate that the entity is in 'constructed' state, which indicates that it got reloaded.
+                    EXPECT_EQ(entity->GetState(), AZ::Entity::State::Constructed);
+                    AZStd::vector<AZ::Component*> entity1Components = entity->GetComponents();
+                    EXPECT_TRUE(entity1Components.size() == 1);
```

Comment:
This should use EXPECT_EQ for better error message.
EXPECT_TRUE wouldn't be able to print how many components the entity1Components vector contains when a failure occurs

EXPECT_TRUE/EXPECT_FALSE should only be used when the singular expression returns a bool or a type convertible to.
If a comparison operator is being used in an EXPECT_TRUE/EXPECT_FALSE statement such as `!`, `==`, `!=`, `<`, etc....
Then it is the incorrect macro to use.
i.e 
```
bool FuncThatReturnsBool()
EXPECT_TRUE(FuncThatReturnsBool()); // This is fine

int FuncThatReturnsInt()
EXPECT_TRUE(FuncThatReturnsInt() == 1); // Inhibits outputting of function return value when the test fails.
```

```suggestion
                    EXPECT_EQ(1, entity1Components.size());
```